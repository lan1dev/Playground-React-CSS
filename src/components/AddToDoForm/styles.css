.todo-input {
  height: 46px;
  width: 300px;
  font-size: 18px;
}

.todo-input:focus {
  outline: none;
}
