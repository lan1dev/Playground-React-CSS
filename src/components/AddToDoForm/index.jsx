import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';
import Button from '../Button';

const propTypes = {
  addToDo: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  toDoValue: PropTypes.string.isRequired
};

const AddToDoForm = ({ addToDo, handleChange, toDoValue }) => (
  <form onSubmit={addToDo}>
    <input className="todo-input" name="todo" onChange={handleChange} value={toDoValue} />
    <Button>Add</Button>
  </form>
);

AddToDoForm.propTypes = propTypes;
export default AddToDoForm;
