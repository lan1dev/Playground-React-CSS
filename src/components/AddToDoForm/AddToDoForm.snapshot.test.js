import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddToDoForm from './index';

describe('AddToDoForm', () => {
  let props;
  let wrapper;

  const Wrapper = () => {
    if (!wrapper) {
      wrapper = shallow(<AddToDoForm />);
    }
    return wrapper;
  };

  // Reset Component and Props before each test.
  beforeEach(() => {
    props = {
      children: undefined
    };
    wrapper = undefined;
  });

  describe('when component has rendered', () => {
    it('Should match snapshot', () => {
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });
});
