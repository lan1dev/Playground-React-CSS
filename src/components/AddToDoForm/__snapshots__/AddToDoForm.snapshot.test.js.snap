// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToDoForm when component has rendered Should match snapshot 1`] = `
<form>
  <input
    className="todo-input"
    name="todo"
  />
  <Button>
    Add
  </Button>
</form>
`;
